app/metrics/jupyterkernelbusyness.go

```diff
@@ -0,0 +1,128 @@
+package metrics
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type JupyterKernelBusynessMetricsHandler struct {
+	logger         *logrus.Logger
+	forwardAddress string
+	listenAddress  string
+	namespace      string
+	serviceName    string
+	instanceName   string
+	metricName     MetricName
+	metric         *prometheus.GaugeVec
+}
+
+func NewJupyterKernelBusynessMetricsHandler(logger *logrus.Logger,
+	forwardAddress string,
+	listenAddress string,
+	namespace string,
+	serviceName string,
+	instanceName string) (*JupyterKernelBusynessMetricsHandler, error) {
+	return &JupyterKernelBusynessMetricsHandler{
+		logger:         logger,
+		forwardAddress: forwardAddress,
+		listenAddress:  listenAddress,
+		namespace:      namespace,
+		serviceName:    serviceName,
+		instanceName:   instanceName,
+		metricName:     JupyterKernelBusynessMetricName,
+	}, nil
+}
+
+func (n *JupyterKernelBusynessMetricsHandler) RegisterMetric() error {
+	gaugeVec := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: string(n.metricName),
+		Help: "Jupyter kernel busyness",
+	}, []string{"namespace", "service_name", "instance_name"})
+
+	if err := prometheus.Register(gaugeVec); err != nil {
+		n.logger.WithError(err).WithField("metricName", string(n.metricName)).Error("Failed to register metric")
+		return err
+	}
+
+	n.logger.WithField("metricName", string(n.metricName)).Info("Metric registered successfully")
+	n.metric = gaugeVec
+
+	return nil
+}
+
+func (n *JupyterKernelBusynessMetricsHandler) CollectData() {
+	ticker := time.NewTicker(5 * time.Second)
+	go func() {
+		for range ticker.C {
+			var kernelsList []interface{}
+			kernelsEndpoint := fmt.Sprintf("http://%s/api/kernels", n.forwardAddress)
+			resp, err := http.Get(kernelsEndpoint)
+			if err != nil {
```

Comment:
See my comment above about error handling. Here for example you're not breaking on error